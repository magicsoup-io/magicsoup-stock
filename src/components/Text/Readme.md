Text:

```jsx
<Text
  fontSize={[ 3, 4, 5 ]}
  fontWeight="bold"
  color="primary">
  This is a basic text element
</Text>
```
