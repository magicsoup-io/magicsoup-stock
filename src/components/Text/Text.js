import styled from 'styled-components'
import {
  fontFamily,
  fontWeight,
  textAlign,
  lineHeight,
  letterSpacing,
} from 'styled-system'

import Box from '../Box'

const themed = key => props => props.theme[key]

const Text = styled(Box)`
  ${fontFamily}
  ${fontWeight}
  ${textAlign}
  ${lineHeight}
  ${letterSpacing}
  ${themed('Text')}
`

/** @component */
export default Text

Text.propTypes = {
  ...fontFamily.propTypes,
  ...fontWeight.propTypes,
  ...textAlign.propTypes,
  ...lineHeight.propTypes,
  ...letterSpacing.propTypes,
}
