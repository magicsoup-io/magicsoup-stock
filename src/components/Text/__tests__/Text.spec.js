import React from 'react'
import renderer from 'react-test-renderer'

import Text from '../Text'

it('renders <Text /> without throwing', () => {
  const tree = renderer.create(<Text> I am a Text</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})
