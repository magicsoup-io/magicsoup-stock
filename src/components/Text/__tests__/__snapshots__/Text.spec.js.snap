// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Text /> without throwing 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH kmMRdh"
>
   I am a Text
</div>
`;
